python/cudf/cudf/core/frame.py

```diff
@@ -1649,13 +1666,16 @@ def _merge(
             else:
                 # left_on == right_on, or different names but same columns
                 # in both cases we can sort by either
-                by = list(to_return[mergeop.left_on]._data.columns)
+                by = [to_return._data[name] for name in mergeop.left_on]
             for i, col in enumerate(by):
                 to_sort[i] = col
             inds = to_sort.argsort()
-            to_return = to_return.take(
-                inds, keep_index=(left_index or right_index)
-            )
+            if isinstance(to_return, cudf.Index):
+                to_return = to_return.take(inds)
```

Comment:
Is this because `Index.take` doesn't accept `keep_index` as a kwarg? I wonder if it's better to just add it there and bring it in line with the other APIs, even if it doesn't affect anything/throws an error.

Reply:
As Index doesn't have any index to itself, the api looks odd with that option.